.dotnet/src/Custom/Chat/DeveloperChatMessage.cs

```diff
@@ -0,0 +1,59 @@
+using System.Collections.Generic;
+
+namespace OpenAI.Chat;
+
+/// <summary>
+/// Represents a chat message of the <c>developer</c> role as supplied to a chat completion request. A developer message is
+/// generally supplied as the first message to a chat completion request and guides the model's behavior across future
+/// <c>assistant</c> role response messages. These messages may help control behavior, style, tone, and
+/// restrictions for a model-based assistant. Developer messages replace system messages for o1 models and newer.
+/// </summary>
+[CodeGenModel("ChatCompletionRequestDeveloperMessage")]
+[CodeGenSuppress("DeveloperChatMessage", typeof(ChatMessageContent))]
+public partial class DeveloperChatMessage : ChatMessage
+{
+    /// <summary>
+    /// Creates a new instance of <see cref="DeveloperChatMessage"/> using a collection of content items.
+    /// For <c>system</c> messages, these can only be of type <c>text</c>.
```

Comment:
system --> developer

Although technically the comment is weird because it's technically referring to the REST API which is an issue carried over from the existing `SystemChatMessage` class. 🙃 I think we can fix that later. For now, changing "system" for "developer" is good enough.